### PR TITLE
upgrade to ColorVectorSpace 0.9 and reduce latency

### PR DESCRIPTION
**Pull request by timholy**
Through better design and precompilation, this shaves about 30% off the
time to run the tests. Of course, the more important consequence is that it
should reflect reductions in "time to first computation" for typical usage.

Perhaps the biggest change, though, is this now loads ColorVectorSpace (the new version 0.9) so that it's always loaded throughout JuliaImages.

closes #158

### CODE REVIEW

**Comment by timholy** [comment]
Test failures are due to `[compat]` bounds in downstream consumers (ReferenceTests, ImageMagick, and ImageInTerminal).

**Comment by Tokazama** [comment]
I think SpatioTemporalTraits.jl is at a point where the next set of changes will probably be tests, updating the README, and changes b/c of feedback. I'm currently trying to get rid of a bunch of dead code elsewhere that ArrayInterface now houses instead (and a whole lot of related experimental bloat  code). Once that's done I should be able to get robust tests for SpatioTemporalTraits, using AxisIndices.jl (which liberally borrows tests from most of the array provides that Images.jl uses). The code is already fairly well optimized because it uses ArrayInterface.jl for getting dimensions and axes across permuted/subset/transposed dimensions.

I'm hoping we can slowly get more packages using ArrayInterface, which will make it so that we don't have to define a bunch of new parametric types for packages like MosaicArrays to work with spatially labelled dimensions. For the time being I'll probably have to replace some calls to `Base.axes` with `ArrayInterface.axes`, but I think we can eventually get to the point where deeply nested dimensional labeled arrays could have working and inferable calls like `reverse(A, dim=is_spatial`.

You could probably start looking at the the code if you want to ensure you at least feel comfortable with the syntax. I was hoping to get too some backend stuff in ArrayInterface that would make it easy to support MappedArrays, but that might have to wait a couple weeks because my schedule is pretty crazy for the next week.

**Comment by johnnychen94** [comment]
For 43c37d5: I've separated the MosaicViews part from this PR into #162; it's totally fine to get that into ImageCore 0.8.x release

**Comment by johnnychen94** [comment]
To smoothly migrate to ColorVectorSpace(CVS) 0.9, I'll separate the task into three steps:

- [x] declare compatibility to CVS in downstream packages
  - [x] https://github.com/JuliaImages/ImageContrastAdjustment.jl/pull/45
  - [x] https://github.com/JuliaImages/ImageMetadata.jl/pull/75
  - [x] https://github.com/JuliaImages/ImageDistances.jl/pull/58
- [x]  load CVS in ImageCore (What this PR does)
- [ ]  remove direct dependency to CVS in downstream packages (would be long term process)

Otherwise, it's a little bit hard to make the CI and Pkg happy.

**Comment by johnnychen94** [comment]
So far the test passes; it seems that this PR is ready but @timholy I'll leave it to you to decide whether it's okay to merge.

To make CI happy, I've restored the version number back to 0.8.22, so we'll need to bump it again to 0.9.0 after merging this PR.

**Review by johnnychen94 — APPROVED** [review]
After separating some of the codes into separated PRs, the changes now look good and safe to me. I'll for a few days and merge this.

File: src/ImageCore.jl
Comment (johnnychen94) [inline src/ImageCore.jl]: https://github.com/JuliaImages/ImageCore.jl/pull/164#issuecomment-824029829

```suggestion
@reexport using ColorVectorSpace
```

**Comment by timholy** [comment]
Thanks for merging this!

**Comment by johnnychen94** [comment]
welcome back 😄

**Comment by timholy** [comment]
Thanks! It's nice to be back. My immediate goal is to get my JuliaCon stuff ready but after that I should slowly catch up. There are something like 8k "non-triagable" notifications so it may take a bit of time.

**Pull request merged**
